templates/skeleton/app/components/Header.tsx

```diff
@@ -48,6 +48,7 @@ export function HeaderMenu({
   publicStoreDomain: HeaderProps['publicStoreDomain'];
 }) {
   const className = `header-menu-${viewport}`;
+  const prefetch = viewport === 'mobile' ? 'viewport' : 'intent';
```

Comment:
Is viewport stable between SSR and initial browser render? A little concerned about hydration miss-matches not knowing the upstream implementations.